oeps/processes/oep-0057-core-repositories.rst

```diff
@@ -0,0 +1,166 @@
+
+* We declare that all access to write, configure, or otherwise administer Core repositories shall be granted through the `Core Contributor Program`_ and managed by its Program Administrators.
+
+* We declare that the addition or removal of repositories from the Core should by reviewed by existing Core Contributors.
+
+  * TODO: Define a review process.
+
+
+Reasoning
+---------
+
+Many other repositories exist as part of the Open edX community. Some of these are forks of Core repositories, and others are upstream repositories in their own right. These various non-Core repositories are vital to different Open edX community members. Furthermore, the existence of non-Core repositories in general is important to the health of the Open edX project, which thrives on collaboration, experimentation, and extension.
+
+However, from these many repositories, we wish to enumerate a smaller set which:
+
+* is a good target for enhancement, documentation, and technical investment;
+
+* defines a logical boundary within which it is fair to enforce Open edX Proposals;
+
+* defines an administrative boundary within which the Core Contributor Program applies;
+
+* makes the Open edX project more comprehensible by collecting its most critical components;
+
+* (TODO: should I add something about tCRIL stewardship/ownership?); and
+
+* encompasses the first-party aspect of the Open edX release supply chain, identifying repositories that need to be:
+
+  * monitoried for discloures under the `Open edX security policy`_,
+  * regularly updated with security patches,
+  * covered by automated unit tests and static analysis, and
+  * ready to deploy to production at any time.
+
+
+Considerations
+--------------
+
+
+* Resolving legacy processes and access policies
+
+  * The process for reviewing addition/removal of repos via the Core Contributor program isn't ready to be spun up yet or even well-defined. In the meantime, tCRIL has been making these judgement calls.
+  * 2U OCM and some of tCRIL retain historical access to many Core repositories outside of the Core Contributor program process. There is a plan to bring those folks under the Core Contributor umbrella, but it will take time.
+  * tCRIL is currently using a few public repositories in the openedx organization for day-to-day work, such as ``openedx/tcril-engineering``. Upon acceptance of this OEP, they would need to make a new GitHub organization for those repositories.
+
+* Cleaning up the Core
+
+  * Many repositories in the openedx GitHub organization do not currently meet the Core criteria. We will need to review the organization contents, and for several repos, transfer them out or invest in them.
+  * Even so, the definition of the Core still captures a large number of repositories, probably larger than we want our "Core" to be. We could reduce it by loosening the package dependencies of many services to being optional instead of strictly required.
```

Comment:
I wonder if it will be valuable to have a second org for the looser but still close to core repos.  For example repos tha aren't required but that are defacto the way you do X if you don't have other opinions?  Even if X plugs into an interface.

The idea of `openedx-contrib` as an org popped into my head as I was reading this so just mentioning it.

Reply:
I think it might be easier to use [topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics) to tier the repos into different categories.

Reply:
I just feel like it would be painful to maintain sync with users, groups, settings and such across a second org. Or dealing with moving repos around between those orgs.

Reply:
We only use topics on a few repos now, and we use them in an inconsistent manner. But we could be more disciplined about it and make it something that folks could reasonably filter on, e.g. https://github.com/orgs/openedx/repositories?q=topic%3Afrontend&type=all&language=&sort=

Reply:
I think it's worth holding off on a specific solution for this to see how things shake out with the maintainers and core changes. It might be that each contributing org maintains its own separate collection, or that the peripheral libraries fall clearly into guilds by topic. I'm just spitballing here but basically I think we should shift these core things and then see what the new emergent patterns are.

Reply:
One small update on this idea--as 2U starts thinking about [pilots for OEP-55](https://github.com/edx/open-source-process-wg/issues/32) I'm realizing that one of our pain points is the way that our internal "ownership" spreadsheet scope is defined, which is basically "anything on the openedx or edx orgs". Arguably that definition will not scale, but I also can't think of a good alternative. Maybe org-based maintainership gets encoded as a tag? e.g. I can filter for all the repos currently maintained by 2U. That would allow us to continue to automate some of our ownership stuff without having to do weird things like duplicate the entire maintainer roster in our spreadsheet (which currently only lists teams internal to 2U).